File: src/main/java/br/senai/saepveterinaria/entity/Produto.java
package br.senai.saepveterinaria.entity;

import br.senai.saepveterinaria.enums.TipoProduto;
import br.senai.saepveterinaria.enums.UnidadeMedida;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.persistence.Version;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDate;

@Data
@Entity
@Table(name = "produto")
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Produto {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer idProduto;

    @Version
    private Long version;

    @NotBlank(message = "O nome é obrigatório")
    @Column(nullable = false)
    private String nome;

    @NotBlank(message = "A marca é obrigatória")
    @Column(nullable = false, length = 100)
    private String marca;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private TipoProduto tipo;

    @Min(value = 0, message = "A quantidade não pode ser negativa")
    @Column(nullable = false)
    private Integer quantidadeEstoque;

    @Min(value = 0, message = "O estoque mínimo não pode ser negativo")
    @Column(nullable = false)
    private Integer estoqueMinimo;

    private LocalDate validade;

    @Column(precision = 10, scale = 2)
    private BigDecimal pesoKg;

    @Column(precision = 10, scale = 2)
    private BigDecimal dosagem;

    @Enumerated(EnumType.STRING)
    @Column(length = 10)
    private UnidadeMedida unidadeMedida;

    @Builder.Default
    @Column(nullable = false, columnDefinition = "boolean default true")
    private Boolean statusProduto = true;
}

File: frontend/src/types/produto.ts
export interface ProdutoCreateDTO {
    nome: string;
    marca: string;
    tipo: string;
    quantidadeEstoque: number;
    estoqueMinimo: number;
    validade: string;
    pesoKg?: number;
    dosagem?: number;
    unidadeMedida: string;
}

export interface ProdutoResponseDTO {
    idProduto: number;
    nome: string;
    marca: string;
    tipo: string;
    quantidadeEstoque: number;
    estoqueMinimo: number;
    validade: string;
    pesoKg?: number;
    dosagem?: number;
    unidadeMedida: string;
}

export interface ProdutoResumoDTO {
    idProduto: number;
    nome: string;
    marca: string;
    tipo: string;
    quantidadeEstoque: number;
    estoqueMinimo: number;
}

export interface ProdutoUpdateDTO {
    nome: string;
    marca: string;
    tipo: string;
    quantidadeEstoque: number;
    estoqueMinimo: number;
    validade: string;
    pesoKg?: number;
    dosagem?: number;
    unidadeMedida: string;
}
